v63005/outref/gtm8877.txt

```diff
@@ -0,0 +1,30 @@
+# NO RESTRICTIONS
+original input
+original input
+
+# FILTER RETURNING -1
+%YDB-E-COMMFILTERERR, Error executing the command filter for gtmzsystem. RESTRICTEDOP
+%YDB-E-RESTRICTEDOP, Attempt to perform a restricted operation: echo 'original input'
+%YDB-E-COMMFILTERERR, Error executing the command filter for gtmpipeopen. RESTRICTEDOP
+%YDB-E-RESTRICTEDOP, Attempt to perform a restricted operation: echo 'original input'
+ZSTATUS=pipeopenfn+4^gtm8877,%YDB-E-IONOTOPEN, Attempt to USE an I/O device which has not been opened
```

Comment:
Why do we see a IONOTOPEN error?

Reply:
The opening of the pipe device is unssucessful due to the filter, and so in the next line it attempts to use a device that had not been opened

Reply:
Then check $test to see if the open succeeded and only if it did, do the use. Otherwise do not do the use. That way we avoid these unrelated errors.

Reply:
As discussed, $test is not set in case OPEN fails and we handle the error with ^incrtrap. So we let the IONOTOPEN error stay in the reference file wherever it shows up.